### DISCUSSION TITLE
How is solidjs better than javascript?

### DISCUSSION THREAD

**Discussion by cscsb** [solidjs/solid#1363] [Q&A]
Since solidjs performance is close to javascript, Why not just use javascript to develop website
, and how solidjs is better than javascript?

**Comment by atk** [comment]
TL; DR: SolidJS scales better.

Hand crafted JS should always be better in terms of performance, since it allows for optimizations that cannot be done automatically.

However, it takes a lot of time to write it, whereas writing code with solid is rather easy, so you'll save a lot of time for near-identical performance and get more maintainable, more reusable code.

**Comment by ghost** [comment]
SolidJS compiler has ability to analyse your code and evaluate some things
Use good practices and stuff

Let's say you have this
```html
<form id="my-form" onsubmit={handleSubmit}>
  <input class="input" oninput={handleInput}>
  <button class="button" onclick={handleClick}>Click Me</button>
<form>
```

In pure js you would write this kinda things
```ts
const $form = document.getElementById('my-form')
const $input =  document.querySelector('.input')
const $button = document.querySelector('.button')

$form.addEventListener('submit', handleSubmit)
$input.addEventListener('input', handleInput)
$button.addEventListener('click', handleClick)
```
Solidjs compiler would write this kinda thing
```ts
// evaluated automatically when component created, no need to go to dom use api to find element
const $form = ... 
// every element has info about its parent,siblings,children
// solid uses that to do this
const $input =  $form.firstChildElement 
const $button = $input.nextSiblingElement 

$form.addEventListener('submit', handleSubmit)
$input.addEventListener('input', handleInput)
$button.addEventListener('click', handleClick)
```

In this example Solid can build a graph of your component and evaluate shortest paths to elements using relative neighbors which is really cool
